icgcget/cli.py

```diff
@@ -76,103 +77,102 @@ def logger_setup(logfile):
     return logger
 
 
-def call_client(args):
-    config = config_parse(args.config)
+def call_client(config, repo, fileid, manifest, output):
```

Comment:
You might want to keep this aggregated in a single object still. Might even be best to have a dedicated class. Probably want to resolve values early before passing downstream. Prevents rework in each of the branches and establishes a contract between methods.